File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltSetTimeScale.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltSetTimeScale extends AltBaseCommand {

  private AltSetTimeScaleParams params;

  public AltSetTimeScale(IMessageHandler messageHandler, AltSetTimeScaleParams params) {
    super(messageHandler);
    this.params = params;
    params.setCommandName("setTimeScale");
  }

  public void Execute() {
    SendCommand(params);
    String data = recvall(params, String.class);
    validateResponse("Ok", data);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/UnityCommands/AltSetTimeScale.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

using Newtonsoft.Json;

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltSetTimeScale : AltBaseCommand
    {
        AltSetTimeScaleParams cmdParams;

        public AltSetTimeScale(IDriverCommunication commHandler, float timeScale) : base(commHandler)
        {
            cmdParams = new AltSetTimeScaleParams(timeScale);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}
